File: demo6b-different-languages/solution/travel-planner-java/src/main/java/io/temporal/ai/workshop/travel/tools/TravelTools.java
// ABOUTME: Travel-planner tools as a dual-annotated interface (Temporal activity + Spring AI tool).
// The @Tool/@ToolParam metadata is what the model sees; @ActivityInterface makes each call durable.
package io.temporal.ai.workshop.travel.tools;

import io.temporal.activity.ActivityInterface;
import io.temporal.activity.ActivityMethod;
import org.springframework.ai.tool.annotation.Tool;
import org.springframework.ai.tool.annotation.ToolParam;

@ActivityInterface
public interface TravelTools {

    @ActivityMethod
    @Tool(
            description =
                    "Get a Wikipedia summary for a topic — useful for cities, countries, regions, attractions.")
    String getWikipediaSummary(
            @ToolParam(description = "The article title to look up, e.g. \"Monaco\", \"Suzuka Circuit\".")
                    String topic);

    @ActivityMethod
    @Tool(
            description =
                    "Get country background information — currency, languages, region, capital, timezones.")
    String getCountryInfo(
            @ToolParam(description = "Country name, e.g. \"Monaco\", \"Italy\", \"Japan\".") String country);
}

File: typescript/decouple-agents/solution/src/shared/travelTools.ts
// ABOUTME: The travel Activities, presented to the model as OpenAI Agents SDK tools.
// Used by the TypeScript travel specialist in challenge 3; challenge 4 replaces it with Python.

import { activityAsTool } from '@temporalio/openai-agents/workflow';
import type { Tool } from '@openai/agents-core';
import type * as activities from './travelActivities';

export function travelTools(): Tool[] {
  const startToCloseTimeout = '30 seconds';

  return [
    activityAsTool<typeof activities.getWikipediaSummary>(
      {
        name: 'getWikipediaSummary',
        description: 'Get a Wikipedia summary for a city, country, region, circuit or landmark.',
        parameters: {
          type: 'object',
          properties: {
            topic: { type: 'string', description: 'Article title, e.g. "Monaco" or "Suzuka Circuit"' },
          },
          required: ['topic'],
          additionalProperties: false,
        },
      },
      { startToCloseTimeout },
    ),
    activityAsTool<typeof activities.getCountryInfo>(
      {
        name: 'getCountryInfo',
        description: 'Get country background: capital, currencies, languages, region and timezones.',
        parameters: {
          type: 'object',
          properties: { country: { type: 'string', description: 'Country name, e.g. "Japan"' } },
          required: ['country'],
          additionalProperties: false,
        },
      },
      { startToCloseTimeout },
    ),
  ];
}
